Elcano_C2_LowLevel/Elcano_C2_LowLevel.ino

```diff
@@ -236,119 +263,156 @@ void setup()
-        //Print7headers(false);
-  //PrintHeaders();
+
+  // @ToDo: Remove any test and calibration code.
+  // //setupWheelRev(); // WheelRev4 addition
+  //CalibrateTurnAngle(32, 20);
+  //calibrationTime_ms = millis();
+
+  attachInterrupt(digitalPinToInterrupt(IRPT_TURN),  ISR_TURN_rise,  RISING);
+  //attachInterrupt(digitalPinToInterrupt(IRPT_RDR),   ISR_RDR_rise,   RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_GO),    ISR_GO_rise,    RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_ESTOP), ISR_ESTOP_rise, RISING);
+  //attachInterrupt(digitalPinToInterrupt(IRPT_RVS),   ISR_RVS_rise,   RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_SWITCH), ISR_SWITCH_rise, RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_MOTOR_FEEDBACK), ISR_MOTOR_FEEDBACK_rise, RISING);
 }
-/*---------------------------------------------------------------------------------------*/
-void loop() {
-  //Serial.println("Start of loop");
-  SerialData Results;
 
-  // Save start time for performance report.
-  unsigned long startTime = micros();
-  // Get the next loop start time.
-  unsigned long nextTime = startTime + LOOP_TIME_US;
+/*---------------------------------------------------------------------------------------*/
 
-  digitalWrite(27, LOW);
-  digitalWrite(33, LOW);
-  startCapturingRCState();
+// Time at which this loop pass should end in order to maintain a
+// loop period of LOOP_TIME_US.
+unsigned long nextTime = micros();
+// Time at which we reach the end of loop(), which should be before
+// nextTime if we have set the loop period long enough.
+unsigned long endTime;
+// How much time we need to wait to finish out this loop pass.
+unsigned long delayTime;
 
-  unsigned long local_results[7];
-  //  PrintDone();
+// Inter-module communications data.
+SerialData Results;
 
-  while (micros() < nextTime &&
-         ~((RC_Done[RC_ESTP] == 1) && (RC_Done[RC_GO] == 1) && (RC_Done[RC_TURN] == 1) && (RC_Done[RC_RDR] == 1)))
-  {
-    ;  //wait
-  }
-  // got data;
-  for (int i = 0; i < 8; i++)
+void loop() {
+  // Get the next loop start time. Note this (and the micros() counter) will
+  // roll over back to zero after they exceed the 32-bit size of unsigned long,
+  // which happens after about 70 minues of operation. If the new nextTime
+  // value is <= LOOP_TIME_US, we've rolled over.
+  nextTime = nextTime + LOOP_TIME_US;
+
+  // ToDo: Verify why this is being copied, e.g. is it to avoid having the
+  // values change while we're using them? Because they can change right here
+  // while we're copying them. Would it help to disable interrupts while we
+  // copy them? or does that just freeze them in some intermediate state?
+  // How about if we just make sure that anytime we exit the interrupt routine,
+  // the RC_elapsed values are in a valid state? Then we don't need to copy
+  // them. (This is all under the assumption that we're copying to avoid
+  // an inconsistent state.)
+  for (int i = 0; i < RC_NUM_SIGNALS; i++)
```

Comment:
Ok, good.  Let me have another look (just paranoia...), and then I'll take out the copy.  I didn't see a way the values could be inconsistent, as they are set in (what is probably) a single store to memory operation at interrupt level, and interrupts are off.  (Note the store to memory might actually be 2-4 stores, depending on the CPU's bus to memory.  But again, interrupts are off when it happens.)